SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -721,22 +761,42 @@ int64 USpatialActorChannel::ReplicateActor()
 		NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID))
 	{
 		if (!NetDriver->LoadBalanceStrategy->ShouldHaveAuthority(*Actor) && !NetDriver->LockingPolicy->IsLocked(Actor))
-		{		
+		{
+			const AActor* NetOwner = Actor->GetNetOwner();
+
+			uint64 HierarchyAuthorityTimestamp = AuthorityTimestamp;
+			if (NetOwner != nullptr)
+			{
+				GetLatestAuthorityChangeFromHierarchy(NetOwner, HierarchyAuthorityTimestamp);
+			}
+
+			const float TimeSinceReceivingAuthInSeconds = double(FPlatformTime::Cycles64() - HierarchyAuthorityTimestamp) * FPlatformTime::GetSecondsPerCycle64();
+			const float MigrationBackoffTimeInSeconds = 1.0f;
```

Comment:
Should this be static?

Reply:
I think this should potentially be exposed in the editor somewhere. When testing this in the example project I was getting confused when my workers weren't changing authority as expected and it was because this backoff was preventing it. That's something I think we can acknowledging affecting here but maybe exposing through editor ( / via lb config?) would provide easier way of testing around it?

Reply:
I agree, I'll put that in a follow up task.

Reply:
Do we want to be storing the Authority change in the SpatialActorChannel or in the LBStrategy? I'm happy to have that in a followup as well.

Reply:
https://improbableio.atlassian.net/browse/UNR-3335 for time setting

Reply:
> Do we want to be storing the Authority change in the SpatialActorChannel or in the LBStrategy? I'm happy to have that in a followup as well.

Despite there not being use for this authority timestamp outside load-balancing, I think it's fine to store it here given how authority role is an Actor-level concept in Unreal.